src/state/opamEnv.ml

```diff
@@ -280,8 +280,8 @@ let get_full
   let updates = u @ updates ~set_opamroot ~set_opamswitch ~force_path st in
   add env0 updates
 
-let is_up_to_date_raw updates =
-  OpamStateConfig.(!r.no_env_notice) ||
+let is_up_to_date_raw ?(check=false) updates =
```

Comment:
I would rename the argument to `skip` instead, and define
```
let is_up_to_date_raw ?(skip=OpamStateConfig.(!r.no_env_notice)) updates = ...
```
I believe it would make things clearer.